woocommerce/woocommerce
# Add date sorting configuration for Analytics

### All Submissions:






### Changes proposed in this Pull Request:

Create new fields `date_paid` and `date_completed` in table `wp_wc_order_stats` (copied from wp_postmeta) and create a configuration in Settings to allow the reports to be based around these two new fields.

It's important to notice that these changes affect not only the Revenue report, but all reports that use `plugins/woocommerce/src/Admin/API/Reports/DataStore.php`. We need to discuss if that's ok.



Closes #35885 .
Closes #35886 .






### How to test the changes in this Pull Request:



TODO: how to test if the `wc_update_740_add_columns_to_order_stats_table` update function and a clean db install works?

1. Have the database schema up-to-date (either a fresh install or run the update function?)
2. Go to Analytics > Revenue
3. Check the report
4. Go to Analytics > Settings
5. Change "Date type" to another value
6. Go to Analytics > Revenue and see if the report has changed (I think that with the way it's setup right now, cache will not work correctly, since the cache key is based around the query arguments, and they are not currently changing)
7. Test for the remaining date type value.



### Other information:




### FOR PR REVIEWER ONLY:
